src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -64,10 +115,20 @@ public static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, ImmutableA
         {
             if (typeSymbol is null)
             {
-                return null;
+                return default;
             }
 
-            return Create(typeSymbol, typeSymbol.IsNullableType(), customModifiers);
+            return CreateNonLazyType(typeSymbol, NonNullTypesTrueContext.Instance, isAnnotated: typeSymbol.IsNullableType(), customModifiers);
+        }
+
+        internal static TypeSymbolWithAnnotations CreateNonLazyType(TypeSymbol typeSymbol, INonNullTypesContext nonNullTypesContext, bool isAnnotated, ImmutableArray<CustomModifier> customModifiers)
+        {
+            return new TypeSymbolWithAnnotations(typeSymbol, nonNullTypesContext, isAnnotated, TypeSymbolWithAnnotationsImplementation.Create(customModifiers));
+        }
+
+        internal static TypeSymbolWithAnnotations CreateLazyNullableType(CSharpCompilation compilation, TypeSymbolWithAnnotations underlying)
+        {
+            return new TypeSymbolWithAnnotations(defaultType: underlying.DefaultType, nonNullTypesContext: null, isAnnotated: true, TypeSymbolWithAnnotationsImplementation.CreateLazy(compilation, underlying));
```

Comment:
>null [](start = 107, length = 4)

The support for `nonNullTypesContext = null` is temporary. Could we avoid adding more scenarios that rely on it?